backend/src/services/__tests__/game.service.test.ts

```diff
@@ -0,0 +1,95 @@
+import { createGame, getGame, validateGameCode } from "../game.service";
+import { Game, GameModel, Setup } from "../../models";
+import mongoose from "mongoose";
+import { PUNCHLINES, SETUPS } from "../../resources";
+import * as Util from "../../util";
+
+beforeAll(async () => {
+  await mongoose.connect(global.__MONGO_URI__, {
+    useNewUrlParser: true,
+  });
+});
+
+afterAll(async () => {
+  await mongoose.connection.close();
+});
+
+describe("createGame Service", () => {
+  let modelSpy: jest.SpyInstance;
+  let codeSpy: jest.SpyInstance;
+
+  beforeEach(() => {
+    modelSpy = jest.spyOn(GameModel.prototype, "save");
+    codeSpy = jest.spyOn(Util, "digitShortCode");
+  });
+
+  afterEach(() => {
+    modelSpy.mockRestore();
+    codeSpy.mockRestore();
+  });
+
+  it("Creates game", async () => {
+    const gameCode = await createGame();
+
+    expect(modelSpy).toHaveBeenCalledTimes(1);
+    const savedGame: Game = modelSpy.mock.instances[0];
+    expect(savedGame.gameCode).toBe(gameCode);
+
+    const setupSort = (a: Setup, b: Setup): number =>
+      a.setup.localeCompare(b.setup);
+    expect(savedGame.punchlines).toHaveLength(PUNCHLINES.length);
+    expect([...savedGame.punchlines]).toEqual(
+      expect.arrayContaining(PUNCHLINES)
+    );
+    expect([...savedGame.punchlines]).not.toEqual(PUNCHLINES);
+    expect(savedGame.setups).toHaveLength(SETUPS.length);
+    expect([...savedGame.setups].sort(setupSort)).toMatchObject(
+      SETUPS.sort(setupSort)
+    );
+    expect([...savedGame.setups]).not.toEqual(SETUPS);
+  });
+
+  it("Mongoose error", async () => {
```

Comment:
I think your test titles could use some slight alterations to be more descriptive. For example `"Mongoose error"` vs `"throws a mongoose error when createGame fails"`

Reply:
I agree - I've improved my test descriptions in 1b6cb1f